json/src/main/scala/com/gu/contentapi/json/Serialization.scala

```diff
@@ -107,66 +109,91 @@ object Serialization {
-
-      val revision = (atom \ "revision").extract[Long]
-      val change = ContentChangeDetails(lastModified, created, published, revision)
-
-      Atom(atomId, atomType, labels, defaultHtml, atomData, change)
+    private def getAtomData(atom: JObject, atomType: AtomType): Option[AtomData] = {
+      atomType match {
+        case AtomType.Quiz => Some(AtomData.Quiz((atom \ "data" \ "quiz").extract[QuizAtom]))
+        case AtomType.Viewpoints => Some(AtomData.Viewpoints((atom \ "data" \ "viewpoints").extract[ViewpointsAtom]))
+        case _ => None
+      }
     }
 
-    // For adding a new atom type define a new function here and then add it to the AtomSerializer
-
-    def getQuizIfExists(rawAtom: JObject): Option[Seq[Atom]] = {
-      Try((rawAtom \ "quizzes").extract[Seq[JObject]]) match {
-        case Success(quizAtoms) =>
-          Some(quizAtoms map { quizAtom =>
-            val atomData = AtomData.Quiz((quizAtom \ "data").extract[QuizAtom])
-            createAtom(quizAtom, AtomType.Quiz, atomData)
-          })
-        case Failure(_) => None
+    private def getAtomTypeFieldName(atomType: AtomType): Option[String] = {
+      atomType match {
+        case AtomType.Quiz => Some("quizzes")
+        case AtomType.Viewpoints => Some("viewpoints")
+        case _ => None
       }
     }
 
-    def getViewpointsIfExists(rawAtom: JObject): Option[Seq[Atom]] = {
-      Try((rawAtom \ "viewpoints").extract[Seq[JObject]]) match {
-        case Success(viewpointAtoms) =>
-          Some(viewpointAtoms map { viewpointAtom =>
-            val atomData = AtomData.Viewpoints((viewpointAtom \ "data").transformField(fixAtomsFields).extract[ViewpointsAtom])
-            createAtom(viewpointAtom, AtomType.Viewpoints, atomData)
-          })
-        case Failure(_) => None
+    //Get an Atom object from the given Atom json
+    def getAtom(rawAtom: JObject): Option[Atom] = {
+      for {
+        atomType <- (rawAtom \ "atomType").extractOpt[AtomType]
+        atomData <- getAtomData(rawAtom, atomType)
+        atom <- getAtom(rawAtom, atomData)
+      } yield atom
+    }
+
+    //Get a list of Atom objects from the given Atoms json for the given atom type
+    def getAtoms(rawAtoms: JObject, atomType: AtomType): Option[Seq[Atom]] = {
+      getAtomTypeFieldName(atomType) flatMap { fieldName =>
+        Try((rawAtoms \ fieldName).extract[Seq[JObject]]).toOption map { atoms =>
+          atoms flatMap (atom => getAtomData(atom, atomType) flatMap (atomData => getAtom(atom, atomData)))
+        }
       }
     }
   }
 
   import AtomHelper._
 
-  object AtomSerializer extends CustomSerializer[Atoms](format => (
+  object DummyAtomObject
+  object AtomsSerializer extends CustomSerializer[Atoms](format => (
+    {
+      case rawAtoms: JObject => Atoms(getAtoms(rawAtoms, AtomType.Quiz), getAtoms(rawAtoms, AtomType.Viewpoints))
+    },
+    {
+      case DummyAtomObject => JString("") //No serialization logic required for Atoms
```

Comment:
Pedantic, but I would amend this comment to be slightly clearer: `// No custom serialization logic required for Atoms`

Also I don't think we need the `DummyAtomObject` hack after all. We can use `PartialFunction.empty[Any, JValue]`. That makes it clearer what we're trying to do here.